zerver/tests/test_invite.py

```diff
@@ -3126,6 +3317,87 @@ def test_create_multiuse_link_with_specified_user_groups_api_call(self) -> None:
             {SystemGroups.MEMBERS, SystemGroups.FULL_MEMBERS},
         )
 
+    def test_edit_multiuse_invite_without_permission_to_subscribe_others(self) -> None:
+        realm = get_realm("zulip")
+
+        # Creating multiuse invite link
+        stream_names = ["Denmark", "Scotland"]
+        streams = [get_stream(stream_name, realm) for stream_name in stream_names]
+
+        stream_ids = [stream.id for stream in streams]
+        multiuse_invite_1 = MultiuseInvite.objects.create(
+            referred_by=self.example_user("iago"),
+            realm=realm,
+        )
+        multiuse_invite_1.streams.set(stream_ids)
+
+        members_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MEMBERS, realm=realm, is_system_group=True
+        )
+        do_change_realm_permission_group_setting(
+            realm, "create_multiuse_invite_group", members_group, acting_user=None
+        )
+        admins_group = NamedUserGroup.objects.get(
+            name=SystemGroups.ADMINISTRATORS, realm=realm, is_system_group=True
+        )
+        do_change_realm_permission_group_setting(
```

Comment:
I guess, this should also test a case where stream level setting allows user to subscribe.